src/coprocessor/codec/data_type/scalar.rs

```diff
@@ -28,6 +28,40 @@ pub enum ScalarValue {
     Json(Option<super::Json>),
 }
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ScalarValueRef<'a> {
+    Int(&'a Option<super::Int>),
+    Real(&'a Option<super::Real>),
+    Decimal(&'a Option<super::Decimal>),
+    Bytes(&'a Option<super::Bytes>),
+    DateTime(&'a Option<super::DateTime>),
+    Duration(&'a Option<super::Duration>),
+    Json(&'a Option<super::Json>),
+}
+
+impl<'a> ScalarValueRef<'a> {
+    #[inline]
+    #[allow(clippy::clone_on_copy)]
+    pub fn to_owned(self) -> ScalarValue {
```

Comment:
The semantic is more like trait `Into` 🤔 ?

Reply:
See https://github.com/tikv/tikv/pull/4786#discussion_r287673754 and https://github.com/tikv/tikv/pull/4786#discussion_r287716581

Reply:
OK, sorry about this.